modules/em/test/test_mrc_indexing.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import print_function
+import IMP
+import IMP.test
+import IMP.em
+import os
+
+
+class Tests(IMP.test.TestCase):
+    """Test that MRC files with non-standard indexing are transposed
+       correctly"""
+
+    def test_centroid_similarity(self):
+        """The centroids of experimental and model-derived map should
+           be similar"""
+        in_mrc = self.get_input_file_name("emd-20362.mrc")
+        in_pdb = self.get_input_file_name("emd-20362.pdb")
+
+        mrw = IMP.em.MRCReaderWriter()
+        self.dmap = IMP.em.read_map(in_mrc, mrw)
+        header = self.dmap.get_header()
+        # Map doesn't set origin correctly; use n*start values instead
+        self.dmap.set_origin(
+            [header.xlen/header.mx * header.nxstart,
+             header.ylen/header.my * header.nystart,
+             header.zlen/header.mz * header.nzstart])
+
+        # Ensure indexing is non-standard
+        self.assertNotEqual((header.mapc, header.mapr, header.maps), (1, 2, 3))
+
+        self.dmap.get_header_writable().resolution = 1.8
+
+        # Make sure that our density map does not get mangled
+        # (e.g. transposed from non-standard to standard indexing again) if we
+        # read it in a second time
+        IMP.em.write_map(self.dmap, 'standard_indexing.mrc')
+        newdmap = IMP.em.read_map('standard_indexing.mrc')
+        self.assertLess(
+            IMP.algebra.get_distance(self.dmap.get_centroid(),
+                                     newdmap.get_centroid()), 5.0)
+        newheader = newdmap.get_header()
+        self.assertEqual((newheader.mapc, newheader.mapr, newheader.maps),
```

Comment:
This line throws an error for me. Seems as if the header written in line 35 retains the original values for `mapc`, `mapr`, and `maps`.

```
        self.assertLess(
  File "/home/saltzberg/development/em_crop_by_particles/imp/modules/em/test/test_mrc_indexing.py", line 42, in test_centroid_similarity
    self.assertEqual((newheader.mapc, newheader.mapr, newheader.maps),
AssertionError: Tuples differ: (2, 1, 3) != (1, 2, 3)
```

Reply:
Hmm, I don't see how that could happen. Are you sure you have all the latest code compiled? The writer doesn't use mapc/r/s; it now always writes (1,2,3)  regardless:
https://github.com/salilab/imp/blob/mrc_indexing/modules/em/src/MRCReaderWriter.cpp#L212-L218

Reply:
Sorry. Yes, I hadn't compiled the new code.